src/input_common/gcadapter/gc_adapter.cpp

```diff
@@ -164,26 +165,27 @@ void Adapter::UpdateYuzuSettings(std::size_t port) {
     for (size_t i = 0; i < pads[port].axis_values.size(); ++i) {
         const s16 value = pads[port].axis_values[i];
 
-        if (value > THRESHOLD || value < -THRESHOLD) {
+        if (value > axis_threshold || value < -axis_threshold) {
             pad_status.axis = static_cast<PadAxes>(i);
             pad_status.axis_value = value;
-            pad_status.axis_threshold = THRESHOLD;
+            pad_status.axis_threshold = axis_threshold;
             pad_queue.Push(pad_status);
         }
     }
 }
 
 void Adapter::UpdateVibrations() {
-    const u8 vibration_states = 8;
+    constexpr u8 vibration_states = 8;
 
     if (!Settings::values.vibration_enabled) {
         // stop all onging vibrations
         if (vibration_counter != vibration_states) {
             vibration_counter = vibration_states;
             vibration_changed = true;
-            for (size_t i = 0; i < pads.size(); ++i) {
-                pads[i].rumble_amplitude = 0;
-                pads[i].enable_vibration = false;
+
+            for (GCController pad : pads) {
```

Comment:
```suggestion
            for (GCController& pad : pads) {
```

Otherwise this is a bug.